src/Symfony/Component/Console/Application.php

```diff
@@ -644,7 +644,7 @@ public function find($name)
 
         // filter out aliases for commands which are already on the list
         if (\count($commands) > 1) {
-            $commandList = $this->commandLoader ? array_merge(array_flip($this->commandLoader->getNames()), $this->commands) : $this->commands;
+            $commandList = $this->commandLoader ? array_merge(array_combine($names = $this->commandLoader->getNames(), array_map(function ($name) { return $this->commandLoader->get($name); }, $names)), $this->commands) : $this->commands;
```

Comment:
We don't need to load all commands here, only matches. Can be done inside the `array_filter` callback below:
```diff
- $commandName = $commandList[$nameOrAlias] instanceof Command ? $commandList[$nameOrAlias]->getName() : $nameOrAlias;
+ if (!$commandList[$nameOrAlias] instanceof Command) {
+     $commandList[$nameOrAlias] = $this->commandLoader->get($nameOrAlias);
+ }
+ $commandName = $commandList[$nameOrAlias]->getName();
```